File: app/src/main/java/com/mindorks/bootcamp/learndagger/di/Module/ApplicationModule.java
package com.mindorks.bootcamp.learndagger.di.Module;

import com.mindorks.bootcamp.learndagger.MyApplication;
import com.mindorks.bootcamp.learndagger.data.local.DatabaseService;
import com.mindorks.bootcamp.learndagger.data.remote.NetworkService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ApplicationModule {

    private MyApplication myApplication;

    public ApplicationModule(MyApplication myApplication) {
        this.myApplication = myApplication;
    }

    @Singleton
    @Provides
    NetworkService provideNetworkService(){
        return new NetworkService(myApplication, "abc");
    }

    @Singleton
    @Provides
    DatabaseService provideDatabaseService(){
        return new DatabaseService(myApplication, "xyz", 1);
    }
}

File: app/src/main/java/com/mindorks/bootcamp/learndagger/di/module/ApplicationModule.kt
package com.mindorks.bootcamp.learndagger.di.module

import android.content.Context
import com.mindorks.bootcamp.learndagger.MyApplication
import com.mindorks.bootcamp.learndagger.di.ApplicationContext
import com.mindorks.bootcamp.learndagger.di.DatabaseInfo
import com.mindorks.bootcamp.learndagger.di.NetworkInfo
import dagger.Module
import dagger.Provides

@Module
class ApplicationModule (var application: MyApplication) {

    @ApplicationContext
    @Provides
    fun provideContext(): Context = application

    @DatabaseInfo
    @Provides
    fun provideDatabaseName(): String = "abc"


    @DatabaseInfo
    @Provides
    fun provideDatabaseVersion(): Int = 1

    @NetworkInfo
    @Provides
    fun provideApiKey():String ="SOME_API_KEY"


}